timlee/networking-with-a-raspberry/README.md

```diff
@@ -0,0 +1,280 @@
+- Server: wlkejf23iojfewooioiwaejf?
+- Client: koijewf2903u0932i9evirj!!!
+
+![TCP handshake](./ssl.png)
+
+The box with an open lock on it in this over-stretched personification is a public key - part of an asymmetric key set. Data can be encrypted by anyone who has the public key (closing the lock) but it can only be decrypted by the person with the private key (opening the lock). In this case, that's the server.
+
+The 'code' generated by the client is a symmetric key - one that can be used to both encrypt and decrypt messages. Once both the client and server have the symmetric key, two-way encryption is achieved.
+
+The public key used in this process is contained within the SSL certificate for a site. The SSL certificate is installed on the web server and any time a client comes along, the web server can provide the certificate to the client so that it can be used during the handshake process.
+
+#### Certificates for site authentication
+
+So you can talk securely to a site calling itself `badger.facts` - great. But what if, without realising it, you're actually communicating with a devious hAXxoR who wants to steal your information?
+
+In addition to containing a public key, an SSL certificate is used to prove a site is who it says it is. Each certificate is _digitally signed_ by a **Certificate Authority** (CA) - a company that specialises in signing certificates to confirm sites are who they say they are. But how can we trust them - who authorises the authorisers?!
+
+Each CA holds their own certificate signed by a **Root Certificate Authority** (root CA) - this is an organisation that is trusted by _The Internet_ (Google, Mozilla, Apple, Microsoft...) to only endorse responsible CAs. This chain of signed notes is used to create a link of trust from the top authority down to each web site.
+
+![Chain of trust](./chain-of-trust.png)
+
+To acquire an SSL certificate you need to prove that you actually own the domain so you can't pretend to be, like, Facebook or something.
+
+So, what does that mean for our internal little site - do we need to prove that we own `badger.facts` so we can get a signed certificate? To do that, we would need to buy the domain name and expose the site to the internet for the CA's to confirm it exists and that we run it. Seems like a lot of work considering the site is only available on our network.
+
+Instead, we can create our own root CA called _Certificates On Me_ (or something) and get all the devices on the network to agree that they trust this CA so that any certificates it signs will be accepted by the browsers.
+
+### Creating your own root CA
+
+There are quite a few steps involved in this, so best if I hand off to an [article I followed](https://deliciousbrains.com/ssl-certificate-authority-for-local-https-development/) to get this configured successfully. This will take you through the steps of:
+
+- configuring your root CA and generating a root CA certificate, which you can do on your Raspberry Pi
```

Comment:
Capital C in ~configuring~ --> Configuring

Reply:
As this is a continuation of the sentence above the list, I think it's ok to stick to lowercase in this case.